collection-scripts/gather_performance_profile_controller

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash
+
+set -x
+
+function ppc_nodes(){
+  
+check_node_gather_pods_ready() {
+   line=$(oc get ds perf-node-gather-daemonset -o=custom-columns=DESIRED:.status.desiredNumberScheduled,READY:.status.numberReady --no-headers -n $NAMESPACE)
+
+    IFS=$' '
+    read desired ready <<< $line
+    IFS=$'\n'
+
+    if [[ "$desired" != "0" ]] && [[ "$ready" == "$desired" ]]
+    then
+       return 0
+    else
+       return 1
+    fi
+}
+
+IFS=$'\n'
+
+BASE_COLLECTION_PATH="/must-gather"
+NODES_PATH=${BASE_COLLECTION_PATH}/nodes
+mkdir -p ${NODES_PATH}
+NAMESPACE_MANIFEST="/etc/performance-profile-node-gather/namespace.yaml"
+SERVICEACCOUNT_MANIFEST="/etc/performance-profile-node-gather/serviceaccount.yaml"
+DAEMONSET_MANIFEST="/etc/performance-profile-node-gather/daemonset.yaml"
+CLUSTER_ROLE_MANIFEST="/etc/performance-profile-node-gather/clusterrole.yaml"
+CLUSTER_ROLE_BINDING_MANIFEST="/etc/performance-profile-node-gather/clusterrolebinding.yaml"
+NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
+# Once you start the pod, the Kubernetes will set the pod hostname to the name of the pod
+# https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
+POD_NAME=${HOSTNAME}
+MUST_GATHER_IMAGE=$(oc get pod -n $NAMESPACE $POD_NAME -o jsonpath="{.spec.containers[0].image}")
+
+POD_IP=$(hostname -I |  tr -d "[:blank:]" )
+echo "[$NAMESPACE/$POD_IP/$POD_NAME]" >> ${BASE_COLLECTION_PATH}/debug
+oc get pod -n $NAMESPACE $POD_NAME -o json >> ${BASE_COLLECTION_PATH}/debug
+
+sed -i -e "s#MUST_GATHER_IMAGE#$MUST_GATHER_IMAGE#" $DAEMONSET_MANIFEST
+sed -i -e "s#NAMESPACE#$NAMESPACE#" $CLUSTER_ROLE_BINDING_MANIFEST
+sed -i -e "s#NAMESPACE#$NAMESPACE#" $DAEMONSET_MANIFEST
+sed -i -e "s#NAMESPACE#$NAMESPACE#" $SERVICEACCOUNT_MANIFEST
+
+oc create -f $SERVICEACCOUNT_MANIFEST
+oc adm policy add-scc-to-user privileged -n $NAMESPACE -z perf-node-gather
+oc create -f $CLUSTER_ROLE_MANIFEST
+oc create -f $CLUSTER_ROLE_BINDING_MANIFEST
+oc create -f $DAEMONSET_MANIFEST
```

Comment:
You can't create any of these, b/c must-gather CLI won't be able to clean it up. You should rather re-use the namespace and the privileged SA created for must-gather pod.